GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -912,42 +905,88 @@ public void PerformRefreshRevisions(Func<RefsFilter, IReadOnlyList<IGitRef>> get
                 _gridView.Enabled = true;
                 _gridView.Focus();
                 _gridView.SelectionChanged += OnGridViewSelectionChanged;
+
+                // Add the spinner controls, removed by SetPage()
+                Controls.Add(_loadingControlSpinner);
+                Controls.Add(_loadingControlText);
+                ShowLoading();
                 _gridView.ResumeLayout();
 
+                IndexWatcher.Reset();
+
+                // Selected is null: Reset the tabs etc.
+                _selectionTimer.Enabled = false;
+                _selectionTimer.Stop();
+                _selectionTimer.Enabled = true;
+                _selectionTimer.Start();
```

Comment:
Nit: Setting `Timer.Enabled` is equivalent to calling `Timer.Stop()` / `.Start`. The (partly redundant) property should only be used for checking whether a timer is running.
```suggestion
                _selectionTimer.Stop();
                _selectionTimer.Start();
```

Reply:
also changed the other occurrences in the file